### ISSUE TITLE
Should `RSVP` be removed from latest docs

### ISSUE THREAD

**Issue by ppcano** [ember-learn/guides-app#298]
The [Why does Ember still use RSVP](https://discuss.emberjs.com/t/readers-questions-why-does-ember-still-use-rsvp/14736) discussion points out the `RSVP` is not needed anymore since the `3.4` version. 

On the latest doc versions, there are a few mentions about `RSVP` that may be unnecessary. Removing them could help to avoid confusion and decrease the number of EmberJS APIs:

- [RSVP.Promise API](https://emberjs.com/api/ember/release/classes/RSVP.Promise)
- [RSVP.EventTarget](https://emberjs.com/api/ember/release/classes/RSVP.EventTarget)

- [Asynchronous routing](https://guides.emberjs.com/release/routing/asynchronous-routing/#toc_a-word-on-promises)
- [Tutorial - component integration test](https://guides.emberjs.com/release/tutorial/autocomplete-component/#toc_an-integration-test)

Besides the documentation status, I wonder if there are a plan or intention to deprecate the `RSVP` API.

**Comment by locks** [comment]
I am confused, you want to remove the documentation to `RSVP` itself?

As for removing `RSVP`, IE11 does not have Promises according to [caniuse](https://caniuse.com/#feat=promises) so we might have to lean on the library a bit longer. I'd also say that `RSVP.hash` and `RSVP.all` are useful, and I'd analyze the cost once we have dead code elimination to have a better sense of its impact.

**Comment by ppcano** [comment]
> I am confused, you want to remove the documentation to RSVP itself?

Yes, this was my suggestion; my reasoning was if `RSVP` is not needed anymore, why does it need to exposed? IMO, it's adding something more to learn without adding much benefit. 

> IE11 does not have Promises

Good catch. I read the [discussion](https://discuss.emberjs.com/t/readers-questions-why-does-ember-still-use-rsvp/14736) and I thought it was totally safe to use. I have to try this [polyfill](https://github.com/stefanpenner/es6-promise#auto-polyfill) in our app.

> I'd also say that RSVP.hash and RSVP.all are useful

`RSVP.all` can be replaced with the native [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) and `RSVP.hash` should be refactored like:

```js
return Promise.all([
  this.store.findAll('stuff'),
  this.store.findAll('otherStuff')
]).then(data => {
  return {
    stuff: data[0],
    otherStuff: data[1]
  };
});
```

The same for other non-compliant `RSVP` methods like `allSettled, hashSettled`.

I didn't know the lack of support on `IE11`, but I still think it is better the documentation promotes the usage of Native Promises and warn about the need of adding a polyfill for IE11 than documenting another library or API. 

Another related point is if or when a `RSVP` deprecation should be created.

**Comment by Gaurav0** [comment]
The biggest issue for me with removing RSVP documentation is that it is not really documented anywhere else besides the github README.md

**Comment by ppcano** [comment]
@Gaurav0 

Agree, the `RSVP` API doc should be accessible, but I think the guides/examples/... could be updated to use Native Promise instead of `RSVP`. 


@locks I have created a [ember-cli-promise-polyfill](https://github.com/loadimpact/ember-cli-promise-polyfill) addon that may be usefull in the future. 

The addon will add the `polyfill` only if the browser Target need it, but it will not be fully useful until Ember builds will not contain `RSVP` anymore, because "at this moment" if the polyfill is appended, two Promise implementations (`RSVP` and `es6-promise`) will be contained in the final build.

I wonder if a RFC would be welcome or this is something more suitable to discuss within the different Ember teams.

**Comment by rwjblue** [comment]
This issue seems to be confusing a few different things at the moment:

* Should we embed the RSVP docs _into_ Ember's API docs?
* Should Ember continue to ship `Ember.RSVP`.

The second is **absolutely** out of scope for this repo, and is a good candidate for an RFC. I do think we will ultimately need a path forward (especially as IE11 usage continues to dwindle), and the only way to do that justice is via an RFC.

Now, that leaves the first point (should https://emberjs.com/api/ directly embed the documentation for RSVP?).  IMHO, not really, but @Gaurav0 is absolutely correct that we need these docs hosted _somewhere_.

I'd propose:

* Standing up a versioned documentation site (perhaps `gh-pages`?) for RSVP itself.
* Updating the Ember API docs to cross link to that new site (at the correct version).

**Comment by wycats** [comment]
@rwjblue There is, at least, one other question, which is whether we could remove incidental uses of RSVP from the guides.

There are some minor (fixable) reasons why we shouldn't do that now, and someone with the details should comment on them in detail.

**Comment by mansona** [comment]
@rwjblue the tech that powers our Ember Guides is now available as an Ember Addon so it is really quite easy for us to stand up a new versioned documentation site with the same sort of structure. Let me know if that works for you and we can discuss getting it up and going 👍

**Comment by miguelcobain** [comment]
Just to add that I created a similar addon for polyfilling `window.Promise`.

It basically sets `window.Promise = Ember.RSVP.Promise`.
I did this to avoid shipping two promises implementations (some polyfill + RSVP).

https://github.com/miguelcobain/ember-cli-promises-polyfill

I think that ideally Ember would work with native Promises and users could polyfill that if needed.
Don't know how far we are from that or what the technical reasons for RSVP are at the moment.
